codersdk/notifications.go

```diff
@@ -280,3 +281,38 @@ func (c *Client) PostTestWebpushMessage(ctx context.Context) error {
 	}
 	return nil
 }
+
+type CustomNotificationContent struct {
+	Title   string `json:"title"`
+	Message string `json:"message"`
+}
+
+type CustomNotificationRequest struct {
+	Content *CustomNotificationContent `json:"content"`
+	// TODO(ssncferreira): Add target (user_ids, roles) to support multi-user and role-based delivery.
+	//   See: https://github.com/coder/coder/issues/19768
+}
+
+func (c CustomNotificationRequest) Validate() error {
+	if c.Content == nil {
+		return xerrors.Errorf("content is required")
+	}
+	if strings.TrimSpace(c.Content.Title) == "" &&
+		strings.TrimSpace(c.Content.Message) == "" {
```

Comment:
I can see an argument for allowing users to send notification messages with just a title. If body is omitted, we can just use title as body if required by the database schema.